agent/agentcontainers/api.go

```diff
@@ -583,8 +647,32 @@ func (api *API) updateContainers(ctx context.Context) error {
 	api.mu.Lock()
 	defer api.mu.Unlock()
 
+	var knownDevcontainers map[string]codersdk.WorkspaceAgentDevcontainer
```

Comment:
Suggestion: Naming so nobody accidentally updates this map rather than `api.knownDevcontainers`. Perhaps `knownDevcontainersBackup` or something along those lines those lines.